rfc/rfc004-verifiable-transactional-graph.md

```diff
@@ -83,19 +83,21 @@ When serializing a reference to string form it MUST be hexadecimal encoded and S
 
 ### 3.3 Ordering, branching and merging
 
-Transactions MUST form a rooted DAG \(Directed Acyclic Graph\) by referring to the previous transaction. This MAY be used to establish _casual ordering_, e.g. registration of a care organization as child object of a vendor. A new transaction MUST be appended to the end of the DAG by referring to the last transaction of the DAG \(_head_\) by including its reference in the **prevs** field.
+Transactions MUST form a rooted DAG \(Directed Acyclic Graph\) by referring to previous transactions. This MAY be used to establish _casual ordering_, e.g. registration of a care organization as child object of a vendor. A new transaction MUST be appended to the end of the DAG by referring to one of the last transactions of the DAG \(_head_\) by including its reference in the **prevs** field.
 
 All transactions referred to by `prevs` MUST be present, since failing to do so would corrupt the DAG.
 
 As the name implies the DAG MUST be acyclic, transactions that introduce a cycle are invalid MUST be ignored. ANY following transaction that refers to the invalid transaction \(direct or indirect\) MUST be ignored as well.
 
-Since it takes some time for the transactions to be synced to all network peers \(eventual consistency\) there COULD be multiple transactions referring to the previous transactions in the **prevs** field, a phenomenon called _branching_. Since branches \(especially old and/or long ones\) may cause transactions to be reordered which hurts performance they MUST be merged as soon as possible. Branches are merged by specifying their heads in the **prevs** field:
+Since it takes some time for the transactions to be synced to all network peers \(eventual consistency\) there COULD be multiple transactions referring to the previous transactions in the **prevs** field, a phenomenon called _branching_. Branching is common and harmless. Branches are merged by specifying their heads in the **prevs** field:
```

Comment:
I discussed with @gerardsn that having many parallel branches might hamper our set reconcilliation algorithm, due to the fact that you can have many, many TXs with the same lamport clock. So with our current protocol, it's not harmless?

Reply:
*relatively harmless*?

Branching will occur nevertheless. Merging them again doesn't have a positive effect on performance.

Reply:
branching is fine as long as new Txs include the most recent LC + 1 as the clock value. If a random branch head is taken as prev to calculate the clock we will have additions to the DAG at any clock level. This against the assumptions made in the set reconciliation algorithm.

So a transaction MUST add the transaction with the highest LC value to its prevs. If there are multiple branches with the same LC, anyone of them is fine. I think this is already covered in these changes, but making this a bit more explicit would help

Reply:
ahh, and the other reason for referencing all prevs was to make sure the transactions used in by the new transaction are all present on the dag. We are already including the Tx containing the key used for signing the new Tx as a prev, are there more Txs containing data that must be present for a new Tx to be processed?

Reply:
> are there more Txs containing data that must be present for a new Tx to be processed?
currently not